filemanager.go

```diff
@@ -0,0 +1,350 @@
+		return err
+	}
+
+	for _, remoteFile := range sortFileMap(m.remoteFiles, false) {
+		fmt.Println(remoteFile.Path)
+	}
+	return nil
+}
+
+// deleteRemoteRecursive deletes all discovered remote files in the filemanager.
+func (m *Manager) deleteRemoteRecursive() error {
+	deletes := sortFileMap(m.remoteFiles, true)
+
+	for _, remoteFile := range deletes {
+		if err := m.deleteRemote(remoteFile); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// sortFileMap returns a slice of files sorted by path
+func sortFileMap(fileMap map[string]File, reverse bool) []File {
+	files := []File{}
+	for _, file := range fileMap {
+		files = append(files, file)
+	}
+	sort.Slice(files, func(i, j int) bool {
+		if reverse {
+			return files[i].Path > files[j].Path
+		}
+		return files[i].Path < files[j].Path
+	})
+	return files
+}
+
+// filterUploads returns the files that need to be uploaded.
+func (m *Manager) filterUploads() (map[string]File, error) {
+
+	uploads := make(map[string]File)
+
+	for localRelativeName, localFile := range m.localFiles {
+		if localFile.IsDir {
+			continue
+		}
+
+		remoteFile, ok := m.remoteFiles[localRelativeName]
+		if !ok {
+			uploads[localRelativeName] = localFile
+			continue
+		}
+
+		if localFile.Size != remoteFile.Size {
+			uploads[localRelativeName] = localFile
+			continue
+		}
+
+		if localDigest, err := getFileDigest(localFile.Path); err != nil {
+			return nil, err
+		} else if localDigest != remoteFile.Digest {
+			uploads[localRelativeName] = localFile
+		}
+	}
+	return uploads, nil
+}
+
+// snapshotLocal returns a channel which receives the infos of the files under the given basePath.
+func (m *Manager) snapshotLocal(localPath string) error {
+
+	stat, err := os.Stat(localPath)
+
+	if err != nil {
+		return err
+	}
+
+	if !stat.IsDir() {
+		return fmt.Errorf("path %s is not a directory or a regular file", localPath)
+	}
+
+	err = filepath.Walk(localPath, func(path string, stat os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		fileName, err := filepath.Rel(localPath, path)
+		if err != nil {
+			return err
+		}
+
+		fileName = filepath.ToSlash(fileName)
+		m.localFiles[fileName] = File{
+			Name:  fileName,
+			Path:  path,
+			Size:  int(stat.Size()),
+			IsDir: stat.IsDir(),
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// listFileManagerFiles returns a channel which receives the infos of the files under the given basePath.
+func (m *Manager) snapshotRemote(ctx context.Context, remotePath string) error {
+
+	searchPath := fmt.Sprintf("^%s/.*", remotePath)
+
+	// Search for the root directory
+	if remotePath == "/" {
+		searchPath = fmt.Sprintf("^%s.*", remotePath)
+	}
+
+	query := ListQuery{
+		ItemsPerPage: 1000,
+		Offset:       0,
+		Search: ListSearch{
+			Path: searchPath,
+		},
+	}
+
+	for {
+		files, err := m.client.ListFiles(ctx, query)
+		if err != nil {
+			return err
+		}
+		for _, file := range files.Items {
+
+			if err != nil {
```

Comment:
This seems unnecessary. Is this perhaps a left over from the Walk function implementation?